core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/ExternalDatabaseService.java

```diff
@@ -154,6 +172,28 @@ public void upgradeDatabase(ClusterView cluster, UpgradeTargetMajorVersion targe
             }
     }
 
+    private void pollUntilUpgradeFlowFinished(String databaseCrn, FlowIdentifier flowIdentifier) {
+        Boolean success = Polling.waitPeriodly(DB_POLLING_CONFIG.getSleepTime(), DB_POLLING_CONFIG.getSleepTimeUnit())
+                .stopIfException(DB_POLLING_CONFIG.getStopPollingIfExceptionOccured())
+                .stopAfterDelay(DB_POLLING_CONFIG.getTimeout(), DB_POLLING_CONFIG.getTimeoutTimeUnit())
+                .run(() -> pollFlowState(flowIdentifier));
+        if (!success) {
+            String message = String.format("Upgrade database flow failed in RedBeams. Database crn: %s, upgrade flow: %s",
+                    databaseCrn, flowIdentifier);
+            LOGGER.warn(message);
+            throw new CloudbreakServiceException(message);
+        }
+    }
+
+    private AttemptResult<Boolean> pollFlowState(FlowIdentifier flowIdentifier) {
+        FlowCheckResponse flowState = redbeamsClient.hasFlowRunningByFlowId(flowIdentifier.getPollableId());
+        LOGGER.debug("Database upgrade polling has active flow: {}, with latest fail: {}",
+                flowState.getHasActiveFlow(), flowState.getLatestFlowFinalizedAndFailed());
+        return flowState.getHasActiveFlow()
+                ? AttemptResults.justContinue()
+                : AttemptResults.finishWith(!flowState.getLatestFlowFinalizedAndFailed());
```

Comment:
Please cover the different code paths with UTs.

Reply:
All code branches are not covered with UTs